### DISCUSSION TITLE
Week 8 Report

### DISCUSSION THREAD

**Discussion by HowDoIUseGit** [SE-TINF22B6/CookPit#35] [General]
# Cookpit ASR analysis and utility tree

## Overview
We have compiled the 13 quality attributes shown in the picture.

![grafik](https://github.com/SE-TINF22B6/CookPit/assets/73703863/14d95aab-5bd2-4b25-ab1a-62bd8e8313aa)

With a special focus on the following 3:
- **Integrability**
  - Accessibility for the Disabled by adapting our color scheme to remove any confusion people with red green weakness could've been experiencing
- **Modifiability**
  - Usage of component-based features and implementations to improve the reusability as well as the introduction to new components
- **Security**
  - Validating the input fields of the login/register fields by implementing a regex based validation process

## Why we chose what we chose
- Inclusive User Experience: Adapting the color scheme for people with red-green color weakness enhances accessibility, ensuring that your website is usable by a wider audience. This inclusivity is not only ethically important but also expands your user base, potentially reaching individuals who might have otherwise faced challenges in accessing or understanding your content.
- Compliance with Standards: Many countries have regulations and guidelines emphasizing web accessibility. By addressing color-related issues, you demonstrate a commitment to accessibility standards, reducing the risk of exclusionary practices and potential legal issues.

- Agile Development: Component-based features enable a more agile development process. By breaking down your website into modular components, you can make modifications or introduce new features without affecting the entire system. This agility is crucial for responding quickly to user feedback, industry trends, or changes in business requirements.
- Reusability: Components can be reused across different parts of your website, promoting consistency and reducing development time. This not only streamlines the coding process but also facilitates easier maintenance. When a change is needed, you can update a single component, and the modifications will propagate throughout the system.
- Scalability: As your website grows, modifiability becomes essential for scalability. The ability to easily modify and extend features ensures that your website can evolve to handle increased traffic, content, and functionality without major overhauls.

- Protecting User Data: Security measures such as input validation are critical for safeguarding user data. Validating login and registration fields using regex helps prevent common security threats like SQL injection or cross-site scripting attacks. By ensuring that only valid and expected input is accepted, you reduce the risk of malicious activities that could compromise user accounts or sensitive information.
- Trust Building: A secure website builds trust with users. Implementing robust security measures demonstrates your commitment to protecting user data, which is especially crucial for login and registration processes where sensitive information is collected. Users are more likely to engage with and return to a website that prioritizes their security.
- Regulatory Compliance: Many regions have stringent data protection regulations. Implementing security measures not only helps protect your users but also ensures compliance with legal requirements. This is increasingly important in an era where data privacy is a top concern for individuals and governments alike.

**Comment by Stev3eM** [comment]
Hello dear CookPit team,

Today you have provided a lot of information. I see that you place a lot of emphasis on performance for the user with your quality attributes, which makes sense. The user experience is always at the forefront and could cost a potential customer before they have even started to explore your app.

I also find it interesting to see that you work with direct time information and I'm curious to see exactly how you implement this and whether the server is well-disposed towards you.

I have one more question about your penultimate point in 'Why we chose what we chose':
You want to build trust with your customer base by presenting the high level of security on the website. How do you plan to prove your high security measures to non-tech-savvy customers?

<br>
**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**
